ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -172,22 +172,36 @@ public void run() {
             if (controlPoint == null) {
                 runnable.run();
             } else {
+                Exception runnableException = null;
                 try {
                     if (controlPoint.beginRequest() == RunResult.RUN) {
                         try {
                             runnable.run();
+                        } catch (Exception e) {
+                            runnableException = e;
+                            throw e;
                         } finally {
                             controlPoint.requestComplete();
                         }
                     } else {
                         throw EeLogger.ROOT_LOGGER.cannotRunScheduledTask(runnable);
                     }
                 } catch (RuntimeException re) {
-                    EeLogger.ROOT_LOGGER.failedToRunTask(re);  // note that this is now a DEBUG message as we rethrow re
-                    throw re;
+                    // WFLY-13043
+                    EeLogger.ROOT_LOGGER.failedToRunTask(runnable,re);
```

Comment:
If we're not going to always rethrow and we're wrapping re when we do rethrow, then I think eLogger.ROOT_LOGGER.failedToRunTask(runnable,re); should be moved into the 'if (runnableException == null) {' and logged at WARN or ERROR, not DEBUG.